spec/log_spec.rb

```diff
@@ -0,0 +1,63 @@
+require File.expand_path(File.join(File.dirname(__FILE__), 'spec_helper'))
+
+describe 'ProgressBar bar log output' do
+  before do
+    Timecop.freeze Time.utc(2010, 3, 10, 0, 0, 0)
+    @progress_bar = ProgressBar.new(100)
+    @progress_bar.stub(:terminal_width) { 60 }
+    Timecop.freeze Time.utc(2010, 3, 10, 0, 0, 10) # 10 seconds later
+
+    @message = "The ships hung in the sky in much the same way that bricks don't."
+    @expected_gap = "     " # this is the computed gap based on the terminal_width and bar
+  end
+
+  subject { @progress_bar.log_s(@message) }
+
+  describe 'at count=0' do
```

Comment:
Probably no reason to re-test all the same things the main spec tests. Maybe just 3 specs that verify the bar gets overwritten correctly when:

1. a normal log message
2. a very long log message
3. a log message containing newlines